### PR TITLE
feat(codegen): add container-codegen make rule

### PR DESCRIPTION
**Pull request by njhale**
### Description
- Adds container-codegen make rule
- Adds golang/protobuf as a tool dep
- Generate protobuf
- Vendor

/assign @ecordell 
/approve

### CODE REVIEW

File: pkg/sqlite/load.go [hunk 1]
```diff
@@ -21,25 +21,25 @@ func NewSQLLiteLoader(outFilename string) (*SQLLoader, error) {
 	}
 
 	createTable := `
-	CREATE TABLE operatorbundle (
+	CREATE TABLE IF NOT EXISTS operatorbundle (
```
Comment (njhale) [inline pkg/sqlite/load.go]: I was seeing some issues after running `make unit` multiple times. An error citing X already exists prompted this quick-fix.

File: pkg/api/grpc_health_v1/health.pb.go [hunk 2]
```diff
@@ -57,17 +46,39 @@ func (x HealthCheckResponse_ServingStatus) String() string {
...
 }
 func (HealthCheckResponse_ServingStatus) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor0, []int{1, 0}
+	return fileDescriptor_health_a4d0607c1c56185f, []int{1, 0}
 }
 
 type HealthCheckRequest struct {
-	Service string `protobuf:"bytes,1,opt,name=service" json:"service,omitempty"`
+	Service              string   `protobuf:"bytes,1,opt,name=service,proto3" json:"service,omitempty"`
+	XXX_NoUnkeyedLiteral struct{} `json:"-"`
```
Comment (njhale) [inline pkg/api/grpc_health_v1/health.pb.go]: This generated code uses a newer version of `protoc-gen-go`.

File: go.sum [hunk 3]
```diff
@@ -167,14 +198,18 @@ gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7/go.mod h1:dt/ZhP58zS4L8KSrWD
 gopkg.in/yaml.v2 v2.2.1 h1:mUhvW9EsL+naU5Q3cakzfE91YhliOondGd6ZrsDBHQE=
 gopkg.in/yaml.v2 v2.2.1/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 honnef.co/go/tools v0.0.0-20180728063816-88497007e858/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
+k8s.io/api v0.0.0-20180904230853-4e7be11eab3f/go.mod h1:iuAfoD4hCxJ8Onx9kaTIt30j7jUFS00AXQi6QMi99vA=
```
[the remark this answers is not in the document]
Comment (njhale) [reply go.sum]: I update the version of OLM `operator-registry` depends on. Running `go mod graph | grep k8s.io` yields:

```
github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20181218235825-c53c51a3572b k8s.io/api@v0.0.0-20181203235848-2dd39edadc55
github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20181218235825-c53c51a3572b k8s.io/apiextensions-apiserver@v0.0.0-20181204003618-e419c5771cdc
github.com/operator-framework/operator-lifecycle-manager@v0.0.0-20181218235825-c53c51a3572b k8s.io/apimachinery@v0.0.0-20181203235515-3d8ee2261517
...
```

[the remark this answers is not in the document]
Comment (ecordell) [reply go.sum]: Right, I'm wondering what changed in OLM. Registry should just require the API types which shouldn't pull in a bunch of kube stuff. But we can revisit later.

**Comment by stevekuznetsov** [comment]
FYI our bots _want_ to merge this but can't

```json
{
  "base-sha": "b16cf547337a1bccae0150f5431df10cc2e0e993",
  "branch": "master",
  "component": "tide",
  "controller": "sync",
  "error": "At least 1 approving review is required by reviewers with write access.",
  "level": "error",
  "msg": "Merge failed: PR is unmergable. How did it pass tests?!",
  "org": "operator-framework",
  "pr": 13,
  "repo": "operator-registry",
  "sha": "71097d915a3b098935ea1184bb3d629e091bae46",
  "time": "2018-12-19T18:31:03Z"
}
```

**Pull request merged**